app/uk/gov/hmrc/customs/inventorylinking/export/controllers/actionbuilders/AuthAction.scala

```diff
@@ -72,17 +74,22 @@ class AuthAction @Inject()(
   // pure function that tames exceptions throw by HMRC auth api into an Either
   // this enables calling function to not worry about recover blocks
   // returns a Future of Left(Result) on error or a Right(AuthorisedRequest) on success
-  private def authoriseAsCsp[A](implicit vhr: ValidatedHeadersRequest[A]): Future[Either[Result, Option[BadgeIdentifier]]] = {
+  private def authoriseAsCsp[A](implicit vhr: ValidatedHeadersRequest[A]): Future[Either[Result, Option[BadgeIdentifierEoriPair]]] = {
     implicit def hc(implicit rh: RequestHeader): HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(rh.headers)
 
```

Comment:
I suggest you look at simplifying this code along the lines of 
```Scala
for {
badgeId <- eitherBadgeIdentifierWithValidation.right
eori <- eitherBadgeIdentifierWithValidation.right
} yield Right(Some(BadgeIdentifierEoriPair(badgeId, eori)))
```
See `HeaderValidator.scala` for an example of the approach